src/Shared/NativeMethodsShared.cs

```diff
@@ -470,6 +501,52 @@ public SystemInformationData()
             }
         }
 
+        /// <summary>
+        /// Get the exact physical core count on Windows
+        /// Useful for getting the exact core count in 32 bits processes,
+        /// as Environment.ProcessorCount has a 32-core limit in that case. 
+        /// </summary>
+        public unsafe static int GetPhysicalCoreCount()
```

Comment:
Ideally this should be copy pasted code from a trustworthy repo (dotnet runtime, corefx, etc) and we just copy paste it and add a comment citing the source. If that's the case, can you please cite the source? (github file line permalink)

Well, ideally ideally it should be in a shared library of some sorts, but let's not get too ideal :)

Reply:
Done https://github.com/dotnet/msbuild/pull/5625/commits/70d41ae6e695ad78acc73543dde8093285ae93b2